[ci] Update uuid to amend vulnerability

My PR (#468) failed the "Security Audit" stage of CI. It seems to be resulting from a recent vulnerability found with uuid, a node module. Unsure if updating will break anything / requires anything else to change, so I'm leaving this as an issue for now. Worth amending as a separate PR. Below is the CI error as it appeared in my PR.

```
Run pnpm audit --ignore-registry-errors
! Corepack is about to download https://registry.npmjs.org/pnpm/-/pnpm-10.30.3.tgz
┌─────────────────────┬────────────────────────────────────────────────────────┐
│ moderate            │ uuid: Missing buffer bounds check in v3/v5/v6 when buf │
│                     │ is provided                                            │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Package             │ uuid                                                   │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Vulnerable versions │ <14.0.0                                                │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Patched versions    │ >=14.0.0                                               │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Paths               │ .>uuid                                                 │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ More info           │ https://github.com/advisories/GHSA-w5hq-g745-h8pq      │
└─────────────────────┴────────────────────────────────────────────────────────┘
1 vulnerabilities found
Severity: 1 moderate
Error: Process completed with exit code 1.
```

--- a/package.json
+++ b/package.json
@@ -43,7 +43,7 @@
     "swagger-jsdoc": "^6.2.8",
     "swagger-ui-express": "^5.0.1",
     "typescript": "~5.9.3",
-    "uuid": "^11.1.0"
+    "uuid": "^14.0.0"
   },
   "pnpm": {
     "overrides": {

--- a/server/model/puzzle.ts
+++ b/server/model/puzzle.ts
@@ -1,7 +1,6 @@
 import crypto from 'crypto';
 import _ from 'lodash';
 import Joi from 'joi';
-import * as uuid from 'uuid';
 import {PuzzleJson, ListPuzzleRequestFilters, AddPuzzleResult} from '@shared/types';
 import {pool} from './pool';
 import {dayOfWeekExtract} from './sql_helpers';
@@ -277,7 +276,7 @@ export async function addPuzzle(
   pid?: string,
   uploadedBy?: string | null
 ): Promise<AddPuzzleResult> {
-  const puzzleId = pid || uuid.v4().substr(0, 8);
+  const puzzleId = pid || crypto.randomUUID().substr(0, 8);
   validatePuzzle(puzzle);
   const contentHash = computePuzzleHash(puzzle);
 

--- a/server/model/puzzle.ts
+++ b/server/model/puzzle.ts
@@ -276,7 +276,7 @@ export async function addPuzzle(
   pid?: string,
   uploadedBy?: string | null
 ): Promise<AddPuzzleResult> {
-  const puzzleId = pid || crypto.randomUUID().substr(0, 8);
+  const puzzleId = pid || crypto.randomUUID().slice(0, 8);
   validatePuzzle(puzzle);
   const contentHash = computePuzzleHash(puzzle);
 

